src/components/scheduling/SchedulingConditionsPanel.svelte

```diff
@@ -0,0 +1,76 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { base } from '$app/paths';
+  import { afterUpdate, beforeUpdate } from 'svelte';
+  import { plan } from '../../stores/plan';
+  import { schedulingSpecConditions, selectedSpecId } from '../../stores/scheduling';
+  import GridMenu from '../menus/GridMenu.svelte';
+  import CssGrid from '../ui/CssGrid.svelte';
+  import Panel from '../ui/Panel.svelte';
+  import SchedulingCondition from './conditions/SchedulingCondition.svelte';
+
+  export let gridId: number;
+
+  let activeElement: HTMLElement;
+  let conditionsFilterText: string = '';
+  let filteredSchedulingSpecConditions: SchedulingSpecCondition[] = [];
+
+  $: filteredSchedulingSpecConditions = $schedulingSpecConditions.filter(spec => {
+    const filterTextLowerCase = conditionsFilterText.toLowerCase();
+    const includesName = spec.condition.name.toLocaleLowerCase().includes(filterTextLowerCase);
+    return includesName;
+  });
+
+  // Manually keep focus as scheduling condition elements are re-ordered.
+  // Svelte currently does not retain focus as elements are moved, even when keyed.
+  // See discussion here: https://github.com/sveltejs/svelte/issues/3973
+  beforeUpdate(() => {
+    activeElement = document.activeElement as HTMLElement;
+  });
+
+  afterUpdate(() => {
+    if (activeElement) {
+      activeElement.focus();
+    }
+  });
+</script>
+
+<Panel>
+  <svelte:fragment slot="header">
+    <GridMenu {gridId} title="Scheduling Conditions" />
+  </svelte:fragment>
+
+  <svelte:fragment slot="body">
+    <CssGrid columns="4fr 1fr" gap="5px">
+      <input
```

Comment:
The input here seems to have too much height:
<img width="320" alt="Screen Shot 2022-12-14 at 10 36 58 AM" src="https://user-images.githubusercontent.com/683355/207682978-0a0ac2b4-763a-4396-9749-a252271db83d.png">

Reply:
Disregard - I was on an old version